src/main/java/uk/gov/hmcts/cmc/claimstore/utils/Lazy.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.hmcts.cmc.claimstore.utils;
+
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+
+import java.util.function.Supplier;
+
+@RequiredArgsConstructor(access = AccessLevel.PRIVATE)
+public class Lazy<T> {
+    private final Supplier<T> supplier;
+    private T instance;
```

Comment:
Does this need to be volatile?

Reply:
At the moment, no; only a single thread will be accessing it at a time.
In general, as a class - probably!
Will update.